open_optical_gating/cli/cli.py

```diff
@@ -447,13 +494,19 @@ def clear_state(self):
 
 
     def get_period_state(self, frame):
-        """ State 2 - get period mode (default requires user input).
-        In this mode we obtain a minimum number of frames
-        Determine a period and then return.
+        """ State 2 - get period mode (default behaviour requires user input).     
+        In this mode we obtain a minimum number of frames,
+        determine a period and then return.
         It is assumed that the user (or cli/flask app) then runs
         the select_period function (and updates the state)
         before running analyse again with the new state.
         """
+        # TODO: JT writes: what user input? Why!? I think this function header needs updating - I'm not convinced any of what it says is accurate!
+        # Or perhaps I am making too many assumptions, and this code is a bit less self-sufficient than I had imagined.
+        # TODO: JT writes: I think this is a very odd way to do things (fill the ref_buffer before attempting to determine the period).
+        # Why are you doing it like that? (the loop in establish_indices now makes more sense to me, at least...)
+        # This is not going to work well live, when we want to lock on to a period ASAP, rather than acquiring an unnecessarily long frame buffer first.
+        # (Is any of this due to concern about taking too much time to analyze one frame before the next one arrives...?)

```

Comment:
I think this is entirely due to the fact that this code started 1) from my offline python, which didn't care and then became 2) an emulated system to see if the pi could process things quickly enough before becoming 3) a live system.

Reply:
Raised issue; resolving this conversation.